service-templates/apigw-lambda-svc/v1/pipeline_infrastructure/main.tf

```diff
@@ -1,393 +1,38 @@
-    }
+provider "aws" {
+  region = var.aws_region
 
-    environment_variable {
-      name  = "service_name"
-      value = var.service.name
+  default_tags {
+    tags = {
+      "proton:pipeline" = var.service_name
     }
   }
-
-  source {
-    buildspec = <<EOF
-                {
-                  "version": "0.2",
-                  "phases": {
-                    "install": {
-                      "runtime-versions":
-                      ${tomap({
-    "ruby2.7"       = jsonencode({ "ruby" = "2.7" })
-    "nodejs12.x"    = jsonencode({ "nodejs" = "12.x" })
-    "python3.8"     = jsonencode({ "python" = "3.8" })
-    "java11"        = jsonencode({ "java" = "openjdk11.x" })
-    "dotnetcore3.1" = jsonencode({ "dotnet" : "3.1" })
-})[var.service_instances[0].outputs.LambdaRuntime]},
-                      "commands": [
-                        "pip3 install --upgrade --user awscli",
-                        "echo 'f6bd1536a743ab170b35c94ed4c7c4479763356bd543af5d391122f4af852460  yq_linux_amd64' > yq_linux_amd64.sha",
-                        "wget https://github.com/mikefarah/yq/releases/download/3.4.0/yq_linux_amd64",
-                        "sha256sum -c yq_linux_amd64.sha",
-                        "mv yq_linux_amd64 /usr/bin/yq",
-                        "chmod +x /usr/bin/yq"
-                      ]
-                    },
-                    "pre_build": {
-                      "commands": [
-                        "cd $CODEBUILD_SRC_DIR/${var.pipeline.inputs.code_dir}",
-                        "${var.pipeline.inputs.unit_test_command}"
-                      ]
-                    },
-                    "build": {
-                      "commands": [
-                        "${var.pipeline.inputs.packaging_command}",
-                        "FUNCTION_KEY=$CODEBUILD_BUILD_NUMBER/function.zip",
-                        "aws s3 cp function.zip s3://$bucket_name/$FUNCTION_KEY"
-                      ]
-                    },
-                    "post_build": {
-                      "commands": [
-                        "aws proton --region $AWS_DEFAULT_REGION get-service --name $service_name | jq -r .service.spec > service.yaml",
-                        "yq w service.yaml 'instances[*].spec.lambda_bucket' \"$bucket_name\" > rendered_service.yaml",
-                        "yq w service.yaml 'instances[*].spec.lambda_key' \"$FUNCTION_KEY\" > rendered_service.yaml"
-                      ]
-                    }
-                  },
-                  "artifacts": {
-                    "files": [
-                      "${var.pipeline.inputs.code_dir}/rendered_service.yaml"
-                    ]
-                  }
-                }
-EOF
-
-type = "CODEPIPELINE"
 }
 
-encryption_key = aws_kms_key.pipeline_artifacts_bucket_key.arn
-}
-
-
-resource "aws_codebuild_project" "deploy_project" {
-  for_each = { for instance in var.service_instances : instance.name => instance }
```

Comment:
Due to some shortcomings in how `for_each` is evaluated, its always best to use static keys for the maps that are being iterated over. This will work fine provided the `name` is always a static string, but if someone were to use `aws_<some_resource>.foo.id` for the `name` attribute here, it would throw an ugly, vague error that looks like this https://github.com/hashicorp/terraform/issues/4149